apps/web/src/pages/templates/components/ListProviders.tsx

```diff
@@ -11,165 +13,142 @@ import { stepNames } from '../constants';
 import { ChannelTitle } from './ChannelTitle';
 import { LackIntegrationAlert } from './LackIntegrationAlert';
 
+const getIntegrationIcon = (colorScheme: string, providerId: string) => {
+  return '/static/images/providers/' + colorScheme + '/square/' + providerId + '.svg';
+};
+
+const ListProvidersContainer = styled.div`
+  margin-bottom: 2rem;
+  overflow: hidden;
+`;
+
+const TitleAndButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  color: ${colors.B60};
+  margin-bottom: 0.75rem;
+  font-size: 0.875rem;
+  line-height: 1.25rem;
+`;
+
+const ConfigureButton = styled(Button)`
+  height: 2rem;
+  padding: 0.5rem 1rem;
+`;
+
+const IntegrationButton = styled(UnstyledButton)<
+  { isActive: boolean; isDark: boolean } & { onClick?: MouseEventHandler<HTMLButtonElement> }
+>`
+  width: 100%;
+  padding: 0.5rem 0.75rem;
+  background: ${({ isDark }) => (isDark ? colors.B20 : colors.B98)};
+  border-radius: 0.5rem;
+  margin-bottom: 0.75rem;
+  line-height: 1;
+  opacity: ${({ isActive, isDark }) => (isActive || isDark ? 1 : 0.4)};
+`;
+
+const IntegrationIcon = styled.img<{ isActive: boolean; isDark: boolean }>`
+  height: 1.5rem;
+  max-width: 8.75rem;
+  opacity: ${({ isActive, isDark }) => (isActive || isDark ? 1 : 0.4)};
+`;
+
+const IntegrationStatusStyled = styled(IntegrationStatus)`
+  min-width: 5rem;
+`;
+
 export const ListProviders = ({
   channel,
-  providers,
+  channelProviders,
   setConfigureChannel,
   setProvider,
 }: {
   channel: ChannelTypeEnum;
-  providers: IIntegratedProvider[];
+  channelProviders: IIntegratedProvider[];
   setConfigureChannel: (channel: ChannelTypeEnum) => void;
   setProvider: (provider: IIntegratedProvider) => void;
 }) => {
   const { colorScheme } = useMantineColorScheme();
   const { environment: currentEnvironment } = useEnvController();
+  const containsNovuProvider = useMemo(
+    () =>
+      NOVU_SMS_EMAIL_PROVIDERS.some(
+        (providerId) => providerId === channelProviders.find((provider) => provider.connected)?.providerId
+      ),
+    [channelProviders]
+  );
+  const providersForTheCurrentEnvironment = useMemo(
+    () =>
+      channelProviders.filter((provider) => provider.connected && provider.environmentId === currentEnvironment?._id),
+    [channelProviders, currentEnvironment?._id]
+  );
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** These are more or less "selectors," and could be written and reused as standalone functions outside of this component and file. This will help us from re-inventing the wheel because I imagine we have an instance or two of this implemented elsewhere in the app
‏

Reply:
We don't because they are specific to this case, but we can extract it into more reusable code.